pkg/controller/workspace/provision/deployment.go

```diff
@@ -140,11 +140,11 @@ func getSpecDeployment(
 	creator := workspace.Labels[config.WorkspaceCreatorLabel]
 	commonEnv := env.CommonEnvironmentVariables(workspace.Name, workspace.Status.WorkspaceId, workspace.Namespace, creator)
 	for idx := range podAdditions.Containers {
-		podAdditions.Containers[idx].Env = append(podAdditions.Containers[idx].Env, commonEnv...)
+		podAdditions.Containers[idx].Env = append(commonEnv, podAdditions.Containers[idx].Env...)
```

Comment:
I've reverted that changes and configured che-machine-exec with an argument instead of env var.
There is a dedicated issue to deal with env var interpolation https://github.com/eclipse/che/issues/17136